openrtb_ext/convert_down.go

```diff
@@ -308,3 +309,27 @@ func clear202303Fields(r *RequestWrapper) {
 		}
 	}
 }
+
+// clear202309Fields sets all fields introduced in OpenRTB 2.6-202309 to default values
+// which will cause them to be omitted during json marshal.
+func clear202309Fields(r *RequestWrapper) {
+	r.ACat = nil
+
+	for _, imp := range r.GetImp() {
+		if audio := imp.Audio; audio != nil {
+			audio.DurFloors = nil
+		}
+
+		if video := imp.Video; video != nil {
+			video.DurFloors = nil
+		}
+
+		if pmp := imp.PMP; pmp != nil {
+			for i := range pmp.Deals {
+				pmp.Deals[i].Guar = 0
+				pmp.Deals[i].MinCPMPerSec = 0
+				pmp.Deals[i].DurFloors = nil
+			}
+		}
+	}
+}
```

Comment:
I think this function can be simplified slightly where we nil check the imp members directly instead of assigning them to a variable first. 

```
func clear202309Fields(r *RequestWrapper) {
    r.ACat = nil

    for _, imp := range r.GetImp() {
        if imp.Audio != nil {
            imp.Audio.DurFloors = nil
        }

        if imp.Video != nil {
            imp.Video.DurFloors = nil
        }

        if imp.PMP != nil {
            for i := range imp.PMP.Deals {
                imp.PMP.Deals[i].Guar = 0
                imp.PMP.Deals[i].MinCPMPerSec = 0
                imp.PMP.Deals[i].DurFloors = nil
            }
        }
    }
}
```

Reply:
I had earlier versions of code written that way. I found that when the hierarchy gets deep the code is harder to follow. This is more evident in `clear26Fields`. It might also be more efficient since the variable is only resolved once and reused - although admittedly the compiler may correct for that on its own.

Do you find the way you suggested to be easier to read?

Reply:
I do find my suggestion easier to read for me, but I have no issue keeping the function as it is.